orc8r/cloud/go/obsidian/access/log_decorator.go

```diff
@@ -15,34 +15,22 @@ package access
 
 import (
 	"fmt"
-	"net/http"
 
 	"github.com/labstack/echo"
 )
 
-// LogRequestDecorator closure, appends remote address, URI & certificate CN
-// (if available from the passed http.Request) to every log string
-func LogRequestDecorator(req *http.Request) func(f string, a ...interface{}) string {
-	return func(f string, a ...interface{}) string {
-		if req != nil {
-			f += "; Remote: %s, URI: %s"
-			a = append(a, req.RemoteAddr, req.RequestURI)
-			ccn := req.Header.Get(CLIENT_CERT_CN_KEY)
-			if len(ccn) > 0 {
-				f += ", Cert CN: %s"
-				a = append(a, ccn)
-			}
+// decorate appends remote address, URI, and certificate CN as available from
+// the passed context.
+func decorate(c echo.Context, fmtStr string, args ...interface{}) string {
+	req := c.Request()
```

Comment:
This is implementation of c.Request():
`

func (c *context) Request() *http.Request {
	return c.request
}

`

You should probably check if c is nil